src/main/java/com/minecolonies/coremod/colony/buildings/modules/settings/BuilderModeSetting.java

```diff
@@ -33,9 +37,23 @@ public BuilderModeSetting(final List<String> value, final int curr)
         set(value.get(curr));
     }
 
+    @Override
+    public boolean isActive(final ISettingsModule module)
+    {
+        return module.getBuilding().getColony().getResearchManager().getResearchEffects().getEffectStrength(BUILDER_MODE) > 0;
+    }
+
     @Override
     public boolean isActive(final ISettingsModuleView module)
     {
         return module.getColony().getResearchManager().getResearchEffects().getEffectStrength(BUILDER_MODE) > 0;
     }
```

Comment:
Reduce the duplicate code here, neither methods rely on the passed in module or view, so redirect both please to an internal private method.

Reply:
While the code looks similar, the types in the method chain are not the same.  It would only be possible to move `.getResearchEffects()` and on to a separate method, which didn't seem sufficiently worthwhile to me.  But I can change that if preferred.

Reply:
Nah leave it then..
I hate our view concept....

I checked it out a while ago, we don't really save that much on network bandwidth and it brings us double the hassle and code to maintain, but @Raycoms wants to keep it. So who am I to judge.

Reply:
It's simply because it isn't possible to serialize everything to both sides =D

Reply:
What?
Yes it is -> NBT We already store the entire state in it, write the entire colony to NBT dump it in a packet and send it...... Done.

Reply:
The question is how often is that send, if we send a 1mb colony every tick then it'll require 20mb/sec

Reply:
Regardless of something being temporary or not, we could still use the same implementations / interfaces everywhere...

Reply:
> The question is how often is that send, if we send a 1mb colony every tick then it'll require 20mb/sec

I agree that it should not be a single packet doing the update, but we could write a set of packets which sync the managers individually or when dirty etc.
I was just making a point about the fact that we technically have all data.

My point is not the sync, my point is the dual interfaces and classes for them.